include/enums/enum_convergence_flags.h

```diff
@@ -27,7 +27,7 @@ namespace libMesh {
 /**
  * Linear solver convergence flags (taken from the PETSc flags)
  */
-enum LinearConvergenceReason {
+enum LinearConvergenceReason : int {
```

Comment:
The main reason to do it for all the enums is so that they can be forward declared, allowing us to reduce header file dependencies. In other words,
```
enum Order : int;
```
is a valid forward declaration in C++11, whereas
```
enum Order;
```
is not.

Reply:
That's a great reason!  Could we put it in the comments, though?

Reply:
> Could we put it in the comments, though?

Sure, do you want a comment in every header where we now employ the `: int` syntax? 

I'm planning a follow-up PR that actually implements the forward declarations. I'll also add a required configure test for it, though it should have been supported since GCC 4.6.

Reply:
I generally dislike boilerplate, but I can't think of any "centralized" place we could put such a comment, so in every header is probably the best we can do.

On the other hand, if we're soon to have forward declared enums everywhere, that'll probably be adequate design decision reminder for me.  I'd still prefer the comments too but if you'd rather not then that's okay.